packages/ember-glimmer/tests/utils/shared-conditional-tests.js

```diff
@@ -374,11 +384,56 @@ export class SharedConditionalsTest extends AbstractConditionalsTest {
     this.assertText('T1F2');
   }
 
+  ['@test it maintains DOM stability when condition changes from a truthy to a different truthy value']() {
+    this.renderValues(true);
+
+    this.assertText('T1');
+
+    this.takeSnapshot();
+
+    this.runTask(() => set(this.context, 'cond1', 'hello'));
+
+    this.assertText('T1');
+
+    this.assertInvariants();
+  }
+
+  ['@test it maintains DOM stability when condition changes from a falsy to a different falsy value']() {
+    this.renderValues(false);
+
+    this.assertText('F1');
+
+    this.takeSnapshot();
+
+    this.runTask(() => set(this.context, 'cond1', ''));
+
+    this.assertText('F1');
+
+    this.assertInvariants();
+  }
+}
+
+export class SharedHelperConditionalsTest extends SharedConditionalsTest {
+  renderValues(...values) {
+    let templates = [];
+    let context = {};
+
+    for (let i = 1; i <= values.length; i++) {
+      templates.push(this.templateFor({ cond: `cond${i}`, truthy: `t${i}`, falsy: `f${i}` }));
```

Comment:
Thanks, that makes sense.

The `"it can use falsy param as binding"`, `"it respects isTruthy when object changes"``... tests were direct ports from htmlbars, happened after I had added support for (3) and I didn't really give them much thought.

It's late here, but I'll take a higher level view of all these tests tomorrow, experiment a little and report back